src/main/java/org/prebid/cache/handlers/GetCacheHandler.java

```diff
@@ -51,32 +78,108 @@ public GetCacheHandler(final ReactiveRepository<PayloadWrapper, String> reposito
         val timerContext = metricsRecorder.createRequestContextTimerOptionalForServiceType(this.type)
                 .orElse(null);
 
-        return request.queryParam(ID_KEY).map(id -> {
-            val normalizedId = String.format("%s%s", config.getPrefix(), id);
-            val responseMono = repository.findById(normalizedId)
-                    .transform(CircuitBreakerOperator.of(circuitBreaker))
-                    .timeout(Duration.ofMillis(config.getTimeoutMs()))
-                    .subscribeOn(Schedulers.parallel())
-                    .transform(this::validateErrorResult)
-                    .flatMap(wrapper -> {
-                        if (wrapper.getPayload().getType().equals(PayloadType.JSON.toString())) {
-                            metricsRecorder.markMeterForTag(this.metricTagPrefix, MetricsRecorder.MeasurementTag.JSON);
-                            return builder.createResponseMono(request, MediaType.APPLICATION_JSON_UTF8, wrapper);
-                        } else if (wrapper.getPayload().getType().equals(PayloadType.XML.toString())) {
-                            metricsRecorder.markMeterForTag(this.metricTagPrefix, MetricsRecorder.MeasurementTag.XML);
-                            return builder.createResponseMono(request, MediaType.APPLICATION_XML, wrapper);
-                        } else {
-                            // unhandled media type
-                            return Mono.error(new UnsupportedMediaTypeException(UNSUPPORTED_MEDIATYPE));
-                        }
-                    })
-                    .switchIfEmpty(ErrorHandler.createResourceNotFound(normalizedId));
-            return finalizeResult(responseMono, request, timerContext);
-        }).orElseGet(() -> {
+        return request.queryParam(ID_KEY).map(id -> fetch(request, id, timerContext)).orElseGet(() -> {
             val responseMono = ErrorHandler.createInvalidParameters();
             return finalizeResult(responseMono, request, timerContext);
         });
+    }
+
+    private Mono<ServerResponse> fetch(final ServerRequest request,
+                                       final String id,
+                                       final Timer.Context timerContext) {
+        val cacheUrl = resolveCacheUrl(request);
+
+        val responseMono = StringUtils.contains(cacheUrl, config.getAllowedProxyHost())
+                || StringUtils.contains(cacheUrl, LOCALHOST)
```

Comment:
Can be simplified to  `StringUtils.containsAny(cacheUrl, config.getAllowedProxyHost(), LOCALHOST)`.